### DISCUSSION TITLE
More granular time periods

### DISCUSSION THREAD

**Discussion by ukutaht** [plausible/analytics#1339] [Anything Else]
From email:

> in the queries where the period is required I need to add a custom one, per: hours, minutes and seconds

The `period` parameter could certainly accept a custom range that goes down to minutes and even seconds. The only bit to think about is how it plays with the `interval` parameter in the timeseries endpoint.

**Comment by Vigasaurus** [comment]
I think this will become even more evident after #1366 - as the functionality to click easily down into an hour or even a minute could theoretically be added. I've done a bit of poking into this, and I'm not entirely sure how the elixir query struct would handle storing a range such as this which has `DateTime`s instead of `Date`s - I assume it's just my lack of Elixir knowledge in knowing how this'd best work, but I'd be happy to implement this with some guidance on how to do that bit.

**Reply by ukutaht** [reply]
> elixir query struct would handle storing a range such as this which has DateTimes instead of Dates

Yeah it will be a pretty big change. I don't know all the implications but I guess I would just change it and see what breaks :smile:
